diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38,7 +38,7 @@ ADD_LIBRARY(LibsModule
 )
 
 pybind11_add_module(${PROJECT_NAME} src/pywrap.cpp)
-target_link_libraries(${PROJECT_NAME} LibsModule)
+target_link_libraries(${PROJECT_NAME} PRIVATE LibsModule)
 
 target_compile_definitions(${PROJECT_NAME} PRIVATE VERSION_INFO=${EXAMPLE_VERSION_INFO})
 target_include_directories(${PROJECT_NAME} PRIVATE ${PYBIND11_INCLUDE_DIRS})
@@ -47,4 +47,4 @@ target_link_libraries(${PROJECT_NAME} PRIVATE Eigen3::Eigen)
 # Find Boost
 find_package(Boost REQUIRED COMPONENTS program_options system filesystem)
 include_directories( ${Boost_INCLUDE_DIRS} )
-target_link_libraries(pbs ${Boost_LIBRARIES})
+target_link_libraries(pbs PRIVATE ${Boost_LIBRARIES})
